src/config.py

```diff
@@ -35,48 +32,46 @@ def get_env_variable(var_name: str,
     ------
     KeyError
         If allow_empty is False and the environment variable is empty string or None
-    ValueError
-        If cast_to is not compatible with the value stored.
     """
-    env_var = os.getenv(var_name, default)
+    env_var = os.getenv(var_name)
+    if default is not None and env_var in (None, ""):
+        # Set env_var to default, but do not override empty str with None
+        env_var = default
     if not allow_empty and env_var in (None, ""):
         raise KeyError(f"Environment variable {var_name} not set, and allow_empty is False")
-    if isinstance(env_var, cast_to):
-        return env_var
-    # noinspection PyTypeChecker
-    return _cast_str(env_var, cast_to)
+    return env_var
 
 
-def _cast_str(str_to_cast: str, cast_to: Type[T]) -> T:
+def get_bool_env_variable(var_name: str, default: Optional[bool] = None, allow_empty: bool = False) -> bool:
```

Comment:
You missed a conversion to using this function in `tests/test_digitaltwin/test_setup_environment.py:SetupEnvironmentTest.setupClass()`